### PR TITLE
[script] [hunting-buddy] [2] update flag names and delete them at end

### PR DESCRIPTION
**Pull request by KatoakDR**
### Dependencies
* https://github.com/rpherbig/dr-scripts/pull/5515

### Background
* Part of a series of updates I'm bringing to `hunting-buddy`
* As dependent PRs are merged then this one's diff will get smaller

### Changes
* Qualify flag names with `hunting-buddy-` prefix
* Delete all flags in `before_dying` block

### CODE REVIEW

**Comment by KatoakDR** [comment]
@rpherbig ready for your review, thanks

File: hunting-buddy.lic [hunk 1]
```diff
@@ -412,8 +412,13 @@ before_dying do
     stop_script(script_name) if Script.running?(script_name)
   end
   DRCA.release_cyclics(get_settings.cyclic_no_release)
-  Flags.delete('room-check')
-  Flags.delete('familiar_drag')
+  flags_to_delete = [
+    'hunting-buddy-room-check', # there are multiple flags with this prefix, one per friend
```
Comment (rpherbig) [inline hunting-buddy.lic:416]: Would it be better to have this be an array of RegExp's?

Comment (KatoakDR) [reply hunting-buddy.lic:416]: @rpherbig What are you thinking they'd be? I'm not sure they need to be regular expressions, but we could simplify to deleting all flags that start with `hunting-buddy-` prefix.

Comment (KatoakDR) [reply hunting-buddy.lic]: resolved per https://github.com/rpherbig/dr-scripts/pull/5517/commits/297eac9f2e55f865f03342301748158ea58178c4

**Pull request merged**